content/blog/from-cms-to-contextual.md

```diff
@@ -0,0 +1,424 @@
+
+```bash
+query {
+  getPostsList {
+    edges {
+      node {
+        sys {
+          basename
+        }
+      }
+    }
+  }
+}
+```
+
+If you run this query in the GraphQL client you will see the following returned:
+
+```bash
+{
+  "data": {
+    "getPostsList": {
+      "edges": [
+        {
+          "node": {
+            "sys": {
+              "filename": "dynamic-routing"
+            }
+          }
+        },
+        {
+          "node": {
+            "sys": {
+              "filename": "hello-world"
+            }
+          }
+        },
+        {
+          "node": {
+            "sys": {
+              "filename": "preview"
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+Adding this query to our Blog.
+
+The NextJS starter blog is served on the dynamic route `/pages/posts/[slug].js`. When you open the file you will see a function called `getStaticPaths` at the bottom of the file.
+
+```bash
+export async function getStaticPaths() {
+
+....
+```
+
+Remove all the code inside of this function and we can update it to use our own code. The first step is to add an import to the top of the file to be able interact with our graphql, and remove the `getPostBySlug` and `getAllPosts` imports we won’t be using:
+
+```bash
+//other imports
+.....
+- import { getPostBySlug, getAllPosts } from '../../lib/api'
++ import { staticRequest } from "tinacms";
+```
+
+Inside of the `getStaticPaths` function we can construct our request to our content-api. When making a request we expect a `query` or `mutation` and then `variables` if needed to be passed to the query, here is an example:
+
+`staticRequest({
+  query: '...', // our query
+}),`
+
+“*What does* `staticRequest` do?”
+
+It’s just a helper function which supplies a query to your locally-running GraphQL server, which is started on port `4001`. You can just as easily use `fetch` or an http client of your choice.
+
+We can use the `getPostsList` query from earlier to build our dynamic routes:
+
+```bash
+export async function getStaticPaths() {
+  const postsListData = await staticRequest({
+    query: `
+      query {
+        getPostList {
+          edges {
+            node {
+            sys {
+              filename
+              }
+            }
+          }
+      }
+    }
+    `,
+  })
+  return {
+    paths: postsListData.getPostList.edges.map(edge => ({
+      params: { slug: edge.node.sys.filename },
+    })),
+    fallback: false,
+  }
+}
+```
+
+**Quick break down of `getStaticPaths`**
+
+The `getStaticPaths` code takes the graphql query we created, because it does not require any `variables` we can send down an empty object. In the return functionality we map through each item in the `postsListData.getPostList` and create a slug for each one.
+
+### Creating getStaticProps query
```

Comment:
The heading hierarchy is reversed between this and the next heading (should be H2 > H3)